assets/js/components/ClusterDetails/ChecksResultOverviewNew.test.jsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+
+import { faker } from '@faker-js/faker';
+
+import ChecksResultOverviewNew from './ChecksResultOverviewNew';
+
+describe('ChecksResultOverviewNew component', () => {
+  it('should render a spinner when in loading state', () => {
+    render(<ChecksResultOverviewNew loading={true} />);
+
+    expect(screen.getByRole('alert')).toBeVisible();
+    expect(screen.queryByText('Passing')).not.toBeInTheDocument();
+    expect(screen.queryByText('Warning')).not.toBeInTheDocument();
+    expect(screen.queryByText('Critical')).not.toBeInTheDocument();
+  });
+
+  it('should render an error message', () => {
+    const error = faker.hacker.noun();
+    render(<ChecksResultOverviewNew error={error} />);
+
+    expect(screen.getByText(error)).toBeVisible();
+    expect(screen.queryByText('Passing')).not.toBeInTheDocument();
+    expect(screen.queryByText('Warning')).not.toBeInTheDocument();
+    expect(screen.queryByText('Critical')).not.toBeInTheDocument();
+  });
+
+  it('should display a message if no last execution was found', () => {
+    render(<ChecksResultOverviewNew />);
+
+    expect(screen.getByText('No check results available.')).toBeVisible();
+    expect(screen.queryByText('Passing')).not.toBeInTheDocument();
+    expect(screen.queryByText('Warning')).not.toBeInTheDocument();
+    expect(screen.queryByText('Critical')).not.toBeInTheDocument();
+  });
+
+  it('should display the check results overview', () => {
+    const passing_count = faker.datatype.number();
+    const warning_count = faker.datatype.number();
+    const critical_count = faker.datatype.number();
+
+    const data = {
+      completed_at: faker.date.past().toISOString(),
+      passing_count: passing_count,
+      warning_count: warning_count,
+      critical_count: critical_count,
+      status: 'completed',
+    };
+
+    render(<ChecksResultOverviewNew data={data} />);
+
+    expect(screen.getByText('Passing')).toBeVisible();
```

Comment:
If you want to be certain that the values correspond to the same header, you could maybe do something like:
```
const passing = screen.getByText('Passing');
expect(passing).toBeVisible();
expect(passing.parent().next().getByText(passing_count)).toBeVisible();
```
Or something similar